alchemist/alchemist-maps/src/main/java/at/jku/traces/json/JKUJSONLoader.java

```diff
@@ -23,15 +24,33 @@
 
 import com.google.common.base.Charsets;
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
 
+import it.unibo.alchemist.model.interfaces.GPSPoint;
 import it.unibo.alchemist.model.interfaces.GPSTrace;
 
 /**
  */
 public final class JKUJSONLoader implements Serializable {
 
     private static final long serialVersionUID = 7144531714361675479L;
-    private static final Gson GSON = new Gson();
+    private static final Gson GSON = new GsonBuilder()
+            .registerTypeAdapter(GPSPoint.class, new JsonDeserializer<GPSPoint>() {
+                @Override
+                public GPSPoint deserialize(final JsonElement json, final Type typeOfT, final JsonDeserializationContext context) throws JsonParseException {
+                    final JsonObject obj = json.getAsJsonObject();
```

Comment:
Let's check that it is an object first (same with the doubles), and fail by IllegalStateException otherwise, so that in case of future problems we've got clear indications.